src/redux/types.ts

```diff
@@ -0,0 +1,36 @@
+
+// This type always is used in either store or in component when 
+// the actions is dispatch from component, right?
+export type AppState = {
+    counterReducer:{
+        counter:number,
+        input: number
+      }
+}
+
+// this types always goes to reducer, right???
+export type initialStateType = {
+    counter:number,
+    input:number
+}
```

Comment:
Correct, this is for the reducer.
The AppState is for the store.
You could move this type inside its reducer cause only that reducer needs it

Reply:
yes, i was thinking, but you had mentioned all types to write in types.ts file, so, i did it so.

Reply:
Yes that I forgot to mention 😄